GL-728: Adds handling of redirects to result pages

alexdesi opened the pull request:
### Jira link

https://transformuk.atlassian.net/browse/GL-728

### What?

- [x] Updated the DetermineNextPage service to reflect all of the conditions around applicable exemptions questions being answered
- [x] Updated the return signature of the DetermineNextPage service to just return the URL we should be on
- [x] Changed the DetermineCategories service return signature to include numerical results for the category that applies

### Why?

- This gets us going through the journey to get to the results pages
- This should unblock Sam

alexdesi committed:
GL-728: This PR changes the DetermineNextPage services and makes various changes to fix the journey from the ExemptionQuestion to the result pages.

willfish commented on app/services/green_lanes/determine_next_page.rb:
```diff
@@ -47,3 +49,7 @@ def question_unanswered?(cat_exemptions_apply)
   end
 end
+
+# result.size > 1 -> question page
```
Am guessing you don't want this :smile:

alexdesi replied:
indeed :smile:

willfish commented on config/routes.rb:
```diff
@@ -115,7 +115,7 @@
       resource :applicable_exemptions, only: %i[new create]
       # resource :check_your_answers, only: %i[new create]
-      resource :results, only: %i[show]
+      resources :results, param: :category, only: %i[show]
```
Is this a query/positional param constraint? Not sure we need constraints for routes rendered/managed by us?

alexdesi replied:
This allows us to use:

'/green_lanes/results/1' in the routes
and get the params['category'] == '1'  

for example
---
Not sure about your point, let's chat about it

willfish commented on spec/requests/green_lanes/applicable_exemptions_controller_spec.rb:
```diff
@@ -0,0 +1,44 @@
+    end
+
+    it { is_expected.to have_http_status :redirect }
+    it { is_expected.to redirect_to('/green_lanes/results/1?commodity_code=4114109000&country_of_origin=UA&moving_date=2024-05-29') }
```
:raised_hands:

willfish committed:
GL-728: Working journey

- Handle the query params that are dispersed between the next page service and the applicable exemptions controller
- Backfill a bit of logic around the next page answers
- Small refactoring

willfish committed:
GL-728: Fixes logic for next page

willfish committed:
GL-728: Fixes broken spec

willfish committed:
GL-728: Fixes tests some more

willfish committed:
GL-728: Enable green lanes in beta
